homeassistant/components/arwn/sensor.py

```diff
@@ -30,26 +30,30 @@ def discover_sensors(topic, payload):
             unit = TEMP_FAHRENHEIT
         else:
             unit = TEMP_CELSIUS
-        return ArwnSensor(name, "temp", unit)
+        return ArwnSensor(topic, name, "temp", unit)
     if domain == "moisture":
         name = f"{parts[2]} Moisture"
-        return ArwnSensor(name, "moisture", unit, "mdi:water-percent")
+        return ArwnSensor(topic, name, "moisture", unit, "mdi:water-percent")
```

Comment:
This will also be the same unique ID ?

Reply:
Each unique ID needs to be unique for the combination of entity domain (`sensor`) and providing integration domain (`arwn`)

Reply:
This uses mqtt under the covers. The topic is the mqtt full path. For most classes of sensors that is unique. Would like like 'arwn/temperature/Basement' for instance. The only places where that ends up being a concern is where there are actually more than 1 sensor to be exposed embedded. That's where the rain and wind sensors come into play.